Tribler/Main/Utility/GuiDBHandler.py

```diff
@@ -165,41 +156,34 @@ def wrapper():
 
         wrapper.__name__ = str(name)
 
-        if not self.onSameThread(workerType) or delay:
-            if workerType == "dbThread":
-                task_name = uId
-                if not task_name:
-                    self._auto_counter += 1
-                    task_name = "guidbhandler %d" % self._auto_counter
+        db_thread_wanted = workerType == "dbThread"
+        threadpool_wanted = workerType == "guiTaskQueue"
+        can_run_in_same_thread = (db_thread_wanted and isInIOThread()) or (threadpool_wanted and isInThreadPool())
```

Comment:
Here, the onSameThread method would check if we're on the IOThread if workerType == "dbThread".
But if we're on the IOThread, it doesn't care about the workerType. This reduces latency slightly.